pkg/agent/datamodel/types.go

```diff
@@ -1696,6 +1696,12 @@ type NodeBootstrappingConfiguration struct {
 	SSHStatus                              SSHStatus
 	DisableCustomData                      bool
 	OutboundType                           string
+	EnableIMDSRestriction                  bool
+	// InsertIMDSRestrictionRuleToMangleTable is only checked when EnableIMDSRestriction is true.
+	// When this is true, iptables rule will be inserted to `mangle` table. This is for Linux Cilium
+	// CNI, which will overwrite the `filter` table so that we can only insert to `mangle` table to avoid
+	// our added rule is overwritten by Cilium.
+	InsertIMDSRestrictionRuleToMangleTable bool
```

Comment:
so when we want to restrict IMDS but we're not using cilium, this will be false?

Reply:
yep, exactly the case, rp will check and pass corresponding value.